### PR TITLE
Add unit test stage for mxnet cpu in debug mode

### PR DESCRIPTION
**Pull request by larroy**
## Description ##
I have seen unit test failing in debug mode. This enables a CPU pass of UT with mxnet compiled in debug mode. This would help uncover some undefined behaviour issues.

https://github.com/apache/incubator-mxnet/issues/11973

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -574,6 +585,20 @@ try {
...
     },
+    'Python3: CPU debug': {
+      node('mxnetlinux-cpu') {
+        ws('workspace/ut-python3-cpu') {
+          try {
+            init_git()
+            unpack_lib('cpu_debug')
+            python3_ut('ubuntu_cpu')
+          } finally {
+            collect_test_results_unix('nosetests_unittest.xml', 'nosetests_python3_cpu_unittest.xml')
```
Comment (marcoabreu) [inline Jenkinsfile]: Duplicate result name

File: ci/docker/runtime_functions.sh [hunk 2]
```diff
@@ -345,6 +345,25 @@ build_ubuntu_cpu_openblas() {
...
+build_ubuntu_cpu_cmake_debug() {
+    set -ex
+    pushd .
+    cd /work/build
+    cmake \
+        -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+        -DCMAKE_C_COMPILER_LAUNCHER=ccache \
+        -DUSE_CUDA=OFF \
+        -DUSE_MKL_IF_AVAILABLE=OFF \
+        -DUSE_OPENMP=OFF \
```
Comment (marcoabreu) [inline ci/docker/runtime_functions.sh]: No openmp?

Comment (larroy) [reply ci/docker/runtime_functions.sh]: Shouldn't we test without? They were some linking issues before.

Comment (marcoabreu) [reply ci/docker/runtime_functions.sh]: Sure, for compilation-only that's definitely a good idea. For running the tests, do you think it makes a big difference (that needs to be validated) between enabled and disabled OpenMP?

**Pull request merged**